addons/test_mail/tests/test_message_track.py

```diff
@@ -283,6 +289,39 @@ def test_tracked_compute(self):
             ('partner_email', 'char', 'foo@example.com', 'bar@example.com'),
         ])
 
+    def test_message_track_monetary(self):
+        """ Update a record with a tracked monetary field """
+        # Create companies with different currencies
+        new_company = self.env['res.company'].create({
+            'name': 'new company',
+            'currency_id': 2,
+        })
+
+        # Check if the tracking value have the correct currency and values
+        self.record.write({
+            'revenue': 100,
+        })
+        self.flush_tracking()
+        self.assertEqual(len(self.record.message_ids), 1)
+        # The currency didn't change but the value did
+        self.assertEqual(self.record.message_ids.tracking_value_ids.currency_id.id, 1)
```

Comment:
Maybe store the tracking in a variable to avoid repetition?
`montary_field_tracking = self.record.message_ids.tracking_value_ids[0]`